src/Zio/UPath.cs

```diff
@@ -83,13 +82,13 @@ internal UPath(string path, bool safe)
     /// Gets a value indicating whether this path is empty (<see cref="FullName"/> equals to the empty string)
     /// </summary>
     /// <value><c>true</c> if this instance is empty; otherwise, <c>false</c>.</value>
-    public bool IsEmpty => FullName == string.Empty;
+    public bool IsEmpty => this.FullName?.Length == 0;
```

Comment:
Produces slightly better code while being equivalent (it's similar to how it's implemented in string.IsNullOrEmpty)